report_dynamic/models/report_dynamic.py

```diff
@@ -0,0 +1,429 @@
+                self._get_sample_record(model)
+            except Exception as e:
+                raise ValidationError(
+                    _("Model %s is not applicable for report. Reason: %s")
+                    % (model, str(e))
+                )
+
+    @api.onchange("template_id")
+    def _onchange_template_id(self):
+        """ When template is chosen, define section_ids """
+        for report in self:
+            if not report.is_template and report.template_id and not report.section_ids:
+                report.section_ids = report.template_id.section_ids
+
+    @api.onchange("model_id")
+    def _onchange_model_id(self):
+        self.ensure_one()
+        res = {}
+        model = self.model_id.model
+        if not model:
+            return res
+        try:
+            self.env[model].search_read([], limit=1)
+        except Exception as e:
+            res["warning"] = {
+                "message": _("Model %s is not applicable for report. Reason: %s")
+                % (model, str(e))
+            }
+            self.model_id = self._origin.model_id.id
+        return res
+
+    @api.depends("model_id", "res_id", "template_id")
+    def _compute_resource_ref(self):
+        for rec in self:
+            if rec.is_template or not rec.model_id:
+                rec.resource_ref = False
+                continue
+            sample_record = self._get_sample_record(rec.model_id.model)
+            rec.resource_ref = "%s,%s" % (
+                rec.model_id.model,
+                rec.res_id or sample_record.id,
+            )
+
+    def _inverse_resource_ref(self):
+        for rec in self:
+            if rec.resource_ref:
+                rec.res_id = rec.resource_ref.id
+                rec.model_id = self.env["ir.model"]._get(rec.resource_ref._name)
+            else:
+                rec.res_id = None
+
+    def get_window_actions(self):
+        return self.env["ir.actions.act_window"].search(
+            [
+                ("res_model", "=", "wizard.report.dynamic"),
+                ("binding_model_id", "=", self.model_id.id),
+            ]
+        )
+
+    def _compute_window_action_exists(self):
+        for rec in self:
+            rec.window_action_exists = bool(rec.get_window_actions())
+
+    @api.depends("resource_ref")
+    def _compute_group_by_record_name(self):
+        for rec in self:
+            rec.group_by_record_name = ""
+            if rec.is_template or not rec.resource_ref:
+                continue
+            rec.group_by_record_name = rec.resource_ref.display_name
+
+    def get_template_xml_id(self):
+        self.ensure_one()
+        if not self.wrapper_report_id:
+            # return a default
+            return "web.external_layout"
+        record = self.env["ir.model.data"].search(
+            [("model", "=", "ir.ui.view"), ("res_id", "=", self.wrapper_report_id.id)],
+            limit=1,
+        )
+        return "{}.{}".format(record.module, record.name)
```

Comment:
what happens if I want to use a template without xmlid?

Reply:
Not answered question?